blocks/extensions/__init__.py

```diff
@@ -363,7 +363,7 @@ def __init__(self, **kwargs):
         super(FinishAfter, self).__init__(**kwargs)
 
     def do(self, which_callback, *args):
-        self.main_loop.log.current_row.training_finish_requested = True
+        self.main_loop.log.current_row['training_finish_requested'] = True
```

Comment:
That should be come a upper-cased constant, I guess (`TRAINING_FINISH_REQUESTED`). Something I wanted to do since long, because otherwise typos only show themselves in runtime and also in an unpredictable way.

I am less concerned about reads like `log.status['epochs_done']` because a wrong read usually quickly let's one know about itself, but doing write with plain text strings is really dangerous.